flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/MongoDBSource.java

```diff
@@ -0,0 +1,386 @@
+         */
+        public Builder<T> copyExistingPipeline(String copyExistingPipeline) {
+            this.copyExistingPipeline = copyExistingPipeline;
+            return this;
+        }
+
+        /**
+         * errors.log.enable
+         *
+         * <p>Whether details of failed operations should be written to the log file. When set to
+         * true, both errors that are tolerated (determined by the errors.tolerance setting) and not
+         * tolerated are written. When set to false, errors that are tolerated are omitted.
+         */
+        public Builder<T> errorsLogEnable(boolean errorsLogEnable) {
+            this.errorsLogEnable = errorsLogEnable;
+            return this;
+        }
+
+        /**
+         * errors.tolerance
+         *
+         * <p>Whether to continue processing messages if an error is encountered. When set to none,
+         * the connector reports an error and blocks further processing of the rest of the records
+         * when it encounters an error. When set to all, the connector silently ignores any bad
+         * messages.
+         *
+         * <p>Default: "none" Accepted Values: "none" or "all"
+         */
+        public Builder<T> errorsTolerance(String errorsTolerance) {
+            this.errorsTolerance = errorsTolerance;
+            return this;
+        }
+
+        /**
+         * heartbeat.interval.ms
+         *
+         * <p>The length of time in milliseconds between sending heartbeat messages. Heartbeat
+         * messages contain the post batch resume token and are sent when no source records have
+         * been published in the specified interval. This improves the resumability of the connector
+         * for low volume namespaces. Use 0 to disable.
+         */
+        public Builder<T> heartbeatIntervalMillis(int heartbeatIntervalMillis) {
+            checkArgument(heartbeatIntervalMillis >= 0);
+            this.heartbeatIntervalMillis = heartbeatIntervalMillis;
+            return this;
+        }
+
+        /**
+         * The deserializer used to convert from consumed {@link
+         * org.apache.kafka.connect.source.SourceRecord}.
+         */
+        public Builder<T> deserializer(DebeziumDeserializationSchema<T> deserializer) {
+            this.deserializer = deserializer;
+            return this;
+        }
+
+        /**
+         * The properties of mongodb kafka connector.
+         * https://docs.mongodb.com/kafka-connector/current/kafka-source
+         */
+        public DebeziumSourceFunction<T> build() {
+            Properties props = new Properties();
+
+            props.setProperty(
+                    "connector.class", MongoDBConnectorSourceConnector.class.getCanonicalName());
+            props.setProperty("name", "mongodb_binlog_source");
+
+            props.setProperty(MongoSourceConfig.CONNECTION_URI_CONFIG, checkNotNull(connectionUri));
+            props.setProperty(MongoSourceConfig.DATABASE_CONFIG, checkNotNull(database));
```

Comment:
hi @Jiabao-Sun Some mongodb needs to have an account and password to connect, here does not consider the setting of the account and password?

Reply:
Hi, @XuQianJin-Stars Please use this pattern `mongodb://username:password@localhost/`